commit cf7cdf236b79005d757b628222115447ff0513e1 (refs/changes/95/795/2)
Author: Cody Peter Mello <cody.mello@joyent.com>
Date:   2016-12-24T00:14:38+00:00 (2 years, 10 months ago)
    
    NAPI-369 Can't use URL /networks/admin/ips
    Reviewed by: Robert Mustacchi <rm@joyent.com>

diff --git a/lib/endpoints/networks/common.js b/lib/endpoints/networks/common.js
index a611dae..3e850bb 100644
--- a/lib/endpoints/networks/common.js
+++ b/lib/endpoints/networks/common.js
@@ -38,15 +38,23 @@ function ensureNetworkExists(netKey, req, res, next) {
     mod_net.get(opts, function (err, net) {
         if (err) {
             if (err.name === 'InvalidParamsError') {
-                return next(new mod_restify.ResourceNotFoundError(err,
+                next(new mod_restify.ResourceNotFoundError(err,
                     'network not found'));
+                return;
             }
-            return next(err);
+            next(err);
+            return;
         }
 
+        /*
+         * We copy the actual UUID over the original value, in case it was one
+         * of the allowed symbolic names, like "admin".
+         */
+        req.params[netKey] = net.uuid;
+
         req.params.network = net;
         req._network = net;
-        return next();
+        next();
     });
 }
 
